Add support for centos (and others in the RH family)

miguelaferreira opened the pull request:
Provided that the epel yum repo is configured (adding yum-epel to the runlist is enough), this cookbook can now install x11vnc and configure it as a service on CentOS.

miguelaferreira committed:
Add support for centos (and others in the RH family)

miguelaferreira committed:
Add new line at EOF

miguelaferreira committed:
Remove provider declaration

jgoulah commented on templates/default/x11vnc.conf.erb:
```diff
@@ -2,3 +2,4 @@ start on login-session-start
 script
   start-stop-daemon --start --exec /usr/bin/x11vnc -- -xkb -noxrecord -noxfixes -noxdamage -display :0 <%= @password_opt %> -auth /var/run/lightdm/root/:0 -forever -bg -o /var/log/x11vnc.log
 end script
+
```
why change this file at all? this is a no-op

miguelaferreira replied:
In your last comment you says to add a new line at the end of every file. That's what I did. Tomorrow I'll for it one more go and will remove changes for the original files.—
Miguel

On Wed, Jul 30, 2014 at 6:23 PM, John Goulah notifications@github.com
wrote:

> > @@ -2,3 +2,4 @@ start on login-session-start
> >  script
> >    start-stop-daemon --start --exec /usr/bin/x11vnc -- -xkb -noxrecord -noxfixes -noxdamage -display :0 <%= @password_opt %> -auth /var/run/lightdm/root/:0 -forever -bg -o /var/log/x11vnc.log
> >  end script
> > +
> > 
> > ## why change this file at all? this is a no-op
> > 
> > Reply to this email directly or view it on GitHub:
> > https://github.com/jgoulah/x11vnc-cookbook/pull/4/files#r15594736

jgoulah replied:
not exactly, your original diff was removing the end of line characters which was not great... but I meant only the files you touched, and technically I didn't mean a "new line" or "extra line" at the end of the file but a "line break" at the end of the last line, if that makes sense... (vi editor will do the right thing...)

miguelaferreira replied:
I've reverted the file to the sate it was before I added the new line

miguelaferreira committed:
Add service restart notification

miguelaferreira committed:
Adjust centos service start priority

miguelaferreira committed:
Remove misplaced new line

miguelaferreira committed:
Multiple changes:
- Make service run with the same parameters in centos and debian
- Render x11vnc.conf only for non centos systems
- Add cookbook name to metadata

Also fix some style issues repoted by
foodcritic and rubocop. Replace " by ' when possible,
use strings for indexing instead of symbols,
and use new hash syntax instead of '=>'

miguelaferreira committed:
Removed extra empty line

miguelaferreira commented:
@jgoulah Any chance you can review this pull request? I need to move cookbooks that depend on this one to production and would prefer not to do it with a patched version. Thanks

jgoulah commented:
I just have to patch mine and make sure nothing breaks, won't be too much longer, sorry for the delay

Sauraus commented:
What YUM repository are using that you have x11vnc as a package available on CentOS?

AFAIK 'vnc-server' is the official way of installing a VNC server on CentOS. http://wiki.centos.org/HowTos/VNC-Server

jgoulah commented on attributes/default.rb:
```diff
@@ -1,3 +1,14 @@
-default[:x11vnc] = {
-  :password => ""
+default['x11vnc'] = {
+  password: ''
```
this change is a syntax error on older ruby versions, seems unnecessary even though they are deprecating their support - not sure I see the reason to break backward compatibility?

miguelaferreira replied:
I see the point for backwards compatibility if you need it. However this syntax is Ruby 1.9 and that is not even the latest version. I can rever this change.

jgoulah commented on recipes/default.rb:
```diff
@@ -2,27 +2,43 @@
 
-service "x11vnc" do
-  action [:enable, :start]
-  provider Chef::Provider::Service::Upstart
```
turns out debian needs this provider line...so another thing breaking compatibility.  can you set the provider in your platform_family case statement?  Maybe Chef::Provider::Service::Init::Redhat (not sure what works on centos)

miguelaferreira replied:
I can fix that too.

jgoulah commented:
there seem to be a lot of changes bundled up in this one pull request.  It would be better to have a PR (one squashed commit too) with the logic changes and another with all the linting/syntax changes (which I'm not totally convinced that we need given it breaks on 1.8.7, I know they are deprecating that soon but not sure they have quite yet)

miguelaferreira commented:
@Sauraus: I'm suing EPEL6

miguelaferreira commented:
@jgoulah: I can address the compatibility issues you found, but I would prefer not to go through all the changes again to separate the linting from the rest. Please let me know if that is a problem for you.

jgoulah commented:
@miguelaferreira its fine this go round, in the future it would be good to have more clear pull requests (one major change per req)

the provider line is the only breaking change (on debian), the syntax "fixes" would probably be better to defer to the near future when 1.8.7 support is dropped

Sauraus commented:
@miguelaferreira that's not a standard CentOS yum repo AFAIK, your code does not work on a plain-vanilla CentOS 6.5 install.

Besides why not just use the vnc-server package on CentOS instead? It comes from the standard repos.

miguelaferreira commented:
I will close this PR to break it up and will make two PRs shortly after.
